### PR TITLE
HARP-6181: Zombie Level [0...3] Tiles

### PR DESCRIPTION
**Pull request by okanishchev**
Signed-off-by: Oleksii Kanishchev <49714978+okanishchev@users.noreply.github.com>

Simply clearing tile loader cache when mapview tile cache is being cleared.

### CODE REVIEW

**Comment by Astrak** [comment]
Thanks! This is the issue that was blocking my ticket for a week :)

File: @here/harp-mapview/lib/DataSource.ts [hunk 1]
```diff
@@ -151,6 +151,10 @@ export abstract class DataSource extends THREE.EventDispatcher {
         // to be overloaded by subclasses
     }
 
+    clearCache() {
```
Comment (CraigBerlin) [inline @here/harp-mapview/lib/DataSource.ts]: Please add comments describing the purpose of the method

Comment (okanishchev) [reply @here/harp-mapview/lib/DataSource.ts]: Done. But it should be obvious from the method name.

Comment (CraigBerlin) [reply @here/harp-mapview/lib/DataSource.ts]: True but this is public API in a core base class that will be implemented by users. For these classes we should have documentation, even if it is trivial.

**Comment by okanishchev** [comment]
@heremaps-bot retest this please

**Comment by nzjony** [comment]
hey @okanishchev, I don't quite understand how this changes make a difference. Why does clearing  the cache make a difference to the loading of the text. I tried a revert locally and I couldn't see any difference, I will make a revert and see if I can prove it using the CI

**Comment by okanishchev** [comment]
@nzjony, decoded tiles for [0..3] zoomlevels are cached in tile loaders, and when you clear tile cache, new tiles for that levels are retrieved from tile loader cache, and not from decoder.
This matters when you change map theme - new styles are not applied to already decoded tiles, only to newly decoded.

**Comment by nzjony** [comment]
> This matters when you change map theme - new styles are not applied to already decoded tiles, only to newly decoded.

`setStyleSet` already calls `this.m_tileLoaderCache.clear();`

**Pull request merged**